### PR TITLE
added support for both static and dynamic path

### PR DESCRIPTION
**Pull request by amitsagtani97**
- When installed from install scripts, it will create a dynamic path.
- When using individual scripts, it will pick the static path if the dynamic path doesn't exist.

### CODE REVIEW

File: OCP-4.X/roles/openshift-install/tasks/main.yml [hunk 1]
```diff
@@ -9,23 +9,27 @@
 # * Install cluster
 #
 
+- name: Create dynamic scale-ci-deploy path
+  set_fact:
+    dynamic_deploy_path: "{{ ansible_user_dir }}/scale-ci-{{ openshift_cluster_name }}-{{ platform }}"
+
```
Comment (rsevilla87) [inline OCP-4.X/roles/openshift-install/tasks/main.yml:15]:
```suggestion

- name: Get scale_ci_deploy directory
  set_fact:
    scale_ci_deploy_diretory: "{{ lookup('env', 'dynamic_deploy_path') | default('{{ ansible_user_dir }}/scale-ci-deploy', true) }}"
```

File: OCP-4.X/roles/openshift-install/tasks/main.yml [hunk 2]
```diff
@@ -9,23 +9,27 @@
...
 - name: Create scale-ci-deploy directories
   file:
     path: "{{ item }}"
     state: directory
   with_items:
-    - "{{ ansible_user_dir }}/scale-ci-{{ openshift_cluster_name }}-{{ platform }}/bin"
+    - "{{ dynamic_deploy_path }}/bin"
```
Comment (rsevilla87) [inline OCP-4.X/roles/openshift-install/tasks/main.yml:21]:
```suggestion
    - "{{ scale_ci_deploy_diretory }}/bin"
```

File: OCP-4.X/roles/openshift-install/tasks/main.yml [hunk 3]
```diff
@@ -9,23 +9,27 @@
...
     - "{{ ansible_user_dir }}/.docker"
 
 - name: Download oc client tools
   get_url:
     url: "{{ openshift_client_location }}"
-    dest: "{{ ansible_user_dir }}/scale-ci-{{ openshift_cluster_name }}-{{ platform }}/bin/{{ openshift_client_location | basename }}"
+    dest: "{{ dynamic_deploy_path }}/bin/{{ openshift_client_location | basename }}"
```
Comment (rsevilla87) [inline OCP-4.X/roles/openshift-install/tasks/main.yml:27]:
```suggestion
    dest: "{{ scale_ci_deploy_diretory }}/bin/{{ openshift_client_location | basename }}"
```

File: OCP-4.X/roles/openshift-install/tasks/main.yml [hunk 4]
```diff
@@ -9,23 +9,27 @@
...
 
 - name: Untar and install oc client tools
   shell: |
     set -o pipefail
-    cd {{ ansible_user_dir }}/scale-ci-{{ openshift_cluster_name }}-{{ platform }}/bin
+    cd {{ dynamic_deploy_path }}/bin
```
Comment (rsevilla87) [inline OCP-4.X/roles/openshift-install/tasks/main.yml:32]:
```suggestion
    cd {{ scale_ci_deploy_diretory }}/bin
```

File: OCP-4.X/roles/openshift-install/tasks/main.yml [hunk 5]
```diff
@@ -67,7 +71,7 @@
   - name: Extract openshift-install binary from the payload
     shell: |
       set -o pipefail
-      cd {{ ansible_user_dir }}/scale-ci-{{ openshift_cluster_name }}-{{ platform }}/bin
+      cd {{ dynamic_deploy_path }}/bin
```
Comment (rsevilla87) [inline OCP-4.X/roles/openshift-install/tasks/main.yml:74]:
```suggestion
      cd {{ scale_ci_deploy_diretory }}/bin
```

File: OCP-4.X/roles/openshift-install/tasks/main.yml [hunk 6]
```diff
@@ -107,7 +111,7 @@
   - name: Copy installer into expected directory
     copy:
       src: "{{ workdir }}/bin/openshift-install"
-      dest: "{{ ansible_user_dir }}/scale-ci-{{ openshift_cluster_name }}-{{ platform }}/bin/openshift-install"
+      dest: "{{ dynamic_deploy_path }}/bin/openshift-install"
```
Comment (rsevilla87) [inline OCP-4.X/roles/openshift-install/tasks/main.yml:114]:
```suggestion
      dest: "{{ scale_ci_deploy_diretory }}/bin/openshift-install"
```

Comment (rsevilla87) [reply OCP-4.X/roles/openshift-install/tasks/main.yml:114]: And so on :)

**Review by rsevilla87 — CHANGES_REQUESTED** [review]
Have added some comments

**Review by rsevilla87 — APPROVED** [review]
LGTM!

File: OCP-4.X/clean-on-osp.yml [hunk 7]
```diff
@@ -3,6 +3,10 @@
 # Playbook to uninstall OCP 4 on OSP and clean OSP environment
 #
 
+- name: Get scale_ci_deploy directory
+  set_fact:
+    scale_ci_deploy_diretory: "{{ lookup('env', 'dynamic_deploy_path') | default('scale-ci-deploy', true) }}"
```
Comment (chaitanyaenr) [inline OCP-4.X/clean-on-osp.yml:8]: s/diretory/directory

File: OCP-4.X/clean-on-osp.yml [hunk 8]
```diff
@@ -18,8 +22,8 @@
     - name: Destroy OCP cluster using openshift-install on OSP
       shell: |
         set -o pipefail
-        cd {{ ansible_user_dir }}/scale-ci-{{ openshift_cluster_name }}-{{ platform }}/
-        export OS_CLIENT_CONFIG_FILE={{ ansible_user_dir }}/scale-ci-{{ openshift_cluster_name }}-{{ platform }}/clouds.yml
+        cd {{ ansible_user_dir }}/{{ scale_ci_deploy_diretory }}/
```
Comment (chaitanyaenr) [inline OCP-4.X/clean-on-osp.yml:25]: Won't this have conflicts when we try to install multiple clusters on the same jump host?

Comment (amitsagtani97) [reply OCP-4.X/clean-on-osp.yml]: this scale_ci_deploy_diretory contains the dynamic install address from here - https://github.com/cloud-bulldozer/scale-ci-deploy/blob/fix_dynamic_path/OCP-4.X/roles/openshift-install/tasks/main.yml#L14

**Review by chaitanyaenr — APPROVED** [review]
LGTM

**Pull request merged**